TON Plugin: Implement stake/unstake/pool-info actions

mikirov opened the pull request:
<!-- Use this template by filling in information and copying and pasting relevant items out of the HTML comments. -->

# Relates to

<!-- LINK TO ISSUE OR TICKET -->
https://github.com/elizaOS/eliza/issues/3031

<!-- This risks section must be filled out before the final review and merge. -->

# Risks

<!--
Low, medium, large. List what kind of risks and what could be affected.
-->

# Background

## What does this PR do?
Add actions for staking/unstaking and reading pool info from TON network.

## What kind of change is this?

<!--
Bug fixes (non-breaking change which fixes an issue)
Improvements (misc. changes to existing features)
Features (non-breaking change which adds functionality)
Updates (new versions of included code)
-->

<!-- This "Why" section is most relevant if there are no linked issues explaining why. If there is a related issue, it might make sense to skip this why section. -->
<!--
## Why are we doing this? Any context or related work?
-->

# Documentation changes needed?

<!--
My changes do not require a change to the project documentation.
My changes require a change to the project documentation.
If documentation change is needed: I have updated the documentation accordingly.
-->

<!-- Please show how you tested the PR. This will really help if the PR needs to be retested and probably help the PR get merged quicker. -->

# Testing

## Where should a reviewer start?

## Detailed testing steps

<!--
None: Automated tests are acceptable.
-->

<!--
- As [anon/admin], go to [link]
  - [do action]
  - verify [result]
-->

<!-- If there is a UI change, please include before and after screenshots or videos. This will speed up PRs being merged. It is extra nice to annotate screenshots with arrows or boxes pointing out the differences. -->
<!--
## Screenshots
### Before
### After
-->

<!-- If there is anything about the deployment, please make a note. -->
<!--
# Deploy Notes
-->

<!--  Copy and paste command line output. -->
<!--
## Database changes
-->

<!--  Please specify deploy instructions if there is something more than the automated steps. -->
<!--
## Deployment instructions
-->

<!-- If you are on Discord, please join https://discord.gg/ai16z and state your Discord username here for the contributor role and join us in #development-feed -->
<!--
## Discord username

-->

tcm390 committed:
Merge branch 'develop' into fix/typos

tcm390 committed:
Merge branch 'develop' into tcm-fix-client-scroll

wtfsayo committed:
updated gitignore + removed build files (#3117)

ai16z-demirix committed:
Merge branch 'develop' into tests/plugin-chainbase

shakkernerd committed:
Merge branch 'develop' into tcm-fix-client-scroll

tcm390 committed:
Merge pull request #3115 from elizaOS/tcm-fix-client-scroll

fix: client scroll to bottom

shakkernerd committed:
Merge branch 'develop' into fix-plugin-apro

shakkernerd committed:
Merge pull request #3106 from AIFlowML/fix-plugin-apro

fix: plugin-apro

ai16z-demirix committed:
Merge branch 'develop' into tests/plugin-chainbase

shakkernerd committed:
Merge pull request #3072 from ai16z-demirix/tests/plugin-chainbase

feat (chore): plugin-chainbase test config and coverage

ai16z-demirix committed:
feat (chore): plugin bittensor test config and coverage (#3064)

* plugin-bittensor: package json

* plugin-bittensor: vitest config

* plugin-bittensor: actions, sn34 test

* plugin-bittensor: fact tests

* Update fact.test.ts

* Update package.json

fixing conflict

* Update package.json

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

ai16z-demirix committed:
feat (chore): plugin-bootstrap: test config and test coverage (#3068)

* Update fact.test.ts

* Update goal.test.ts

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

shakkernerd committed:
Merge branch 'develop' into fix/typos

shakkernerd committed:
Merge branch 'develop' into docker-fix

shakkernerd committed:
Merge pull request #3110 from jonathangus/docker-fix

fix: docker build command by skipping eslint files

Commit:
chore(plugin-chainbase): add @biomejs/biome dev dependency

shakkernerd committed:
fix: export dexScreenerPlugin

shakkernerd committed:
Merge pull request #3120 from elizaOS/fix/plugin-dexscreener

fix: export dexScreenerPlugin

shakkernerd committed:
Merge branch 'develop' into fix/plugin-chainbase

shakkernerd committed:
Merge pull request #3118 from akugone/fix/plugin-chainbase

fix:chore(plugin-chainbase): add missing comma in package.json

alexpaden committed:
save imageUrls for outbound tweets/messages

interactions.ts handles inbound
utils.ts handles outbound

shakkernerd committed:
fix: remove duplicate litPlugin import

shakkernerd committed:
Merge pull request #3121 from elizaOS/fix/agent_duplicates

fix: remove duplicate litPlugin import

0xCardiE committed:
add provider, action and template for pools per network

0xCardiE committed:
change names

odilitime committed:
Merge pull request #2361 from elizaOS/develop

chore: dev => main 0.1.9

tcm390 committed:
remove duplicated plugins

john-xina-p88 committed:
Merge branch 'develop' into develop

vipocenka committed:
Update api.ts

tcm390 committed:
Merge branch 'develop' into fix/typos

tcm390 committed:
Merge pull request #3111 from vipocenka/fix/typos

fix: typos in multiple files

v1xingyue committed:
some fix

v1xingyue committed:
Merge branch 'main' into some-fix

tcm390 committed:
Merge branch 'develop' into feature/twitter-messages-image-outbound

tcm390 committed:
Merge pull request #3122 from alexpaden/feature/twitter-messages-image-outbound

feat: save imageUrls for outbound tweets/messages

HashWarlock committed:
fix: upgrade openai and vercel ai packages to fix o1 errors

lalalune committed:
Merge pull request #3146 from elizaOS/fix/openai-o1-fix

fix: upgrade openai and vercel ai packages to fix o1 errors

You-saku committed:
add unwritten text and fix typo in README_JA.md

v1xingyue committed:
update pnpm version in dockerfile

Marcofann committed:
fix ADRESS_SECURITY_CHECK in GoplusSecurityService.ts

Marcofann committed:
fix ADRESS_SECURITY_CHECK in index.ts

Marcofann committed:
fix ADRESS_SECURITY_CHECK in GoPlusManage.ts

Marcofann committed:
fix ADRESS_SECURITY_CHECK in README.md

strmfos committed:
Update chat_2024-12-07.md

strmfos committed:
Update chat_2024-10-27.md

PixelPil0t1 committed:
Update image.yaml

PixelPil0t1 committed:
Update pr.yaml

AIFlowML committed:
Fixed all the issues and added biome

AIFlowML committed:
Merge branch 'develop' into fix-plugin-allors

0xCardiE committed:
Merge branch 'develop' into feat/coingecko_advanced_more_endpoints

AIFlowML committed:
Fixed multiple issues and aded biome.

AIFlowML committed:
Merge branch 'main' into fix-plugin-akash-v1

AIFlowML committed:
Fixed the typing and added biome

AIFlowML committed:
Merge branch 'main' into fix-plugin-agentkit

AIFlowML committed:
fixed all the issues but deployTokenAction.ts for the ANY and the result need probaly a refactoring.

AIFlowML committed:
Merge branch 'develop' into fix-plugin-abstract

AIFlowML committed:
Merge branch 'develop' into fix-plugin-agentkit

0xCardiE committed:
add new pools by network

AIFlowML committed:
Fixed all the issues and added biome

AIFlowML committed:
Merge branch 'develop' into fix-plugin-3g-gen

mikirov committed:
feat: add stake/unstake/pool-info actions

AIFlowML committed:
Fixed all the issues and added Biome

AIFlowML committed:
Merge branch 'develop' into fix-plugin-0x

0xCardiE committed:
add mappings from provider

AIFlowML committed:
Fixed all the issues and added biome. Only issues still in place in upload.ts i had to add an any to overcome the issue on the line 370 const flowContract = getFlowContract(runtime.getSetting(ZEROG_FLOW_ADDRESS), signer as any);

AIFlowML committed:
Merge branch 'develop' into fix-plugin-0g

AIFlowML committed:
Added biome to - plugin-zksync-era - plugin-zilliqa - plugin-zerion - plugin-whatsapp

AIFlowML committed:
Merge branch 'develop' into fix-multi-biome-01

Dashamala91 reviewed (commented):
![1738154355950.jpg](https://github.com/user-attachments/assets/8e1c166a-90f1-4b83-ba57-32a69afc47cf)

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-web-search
- plugin-video-generation
- plugin-udio
- plugin-twitter

shakkernerd committed:
Merge pull request #3180 from AIFlowML/fix-multi-biome-01

fix: multi-biome-01

shakkernerd committed:
Merge branch 'develop' into fix-plugin-0g

AIFlowML committed:
Merge branch 'develop' into fix-multi-biome-02

shakkernerd committed:
Merge pull request #3179 from AIFlowML/fix-plugin-0g

fix: plugin-0g

shakkernerd committed:
Merge branch 'develop' into fix-multi-biome-02

shakkernerd committed:
Merge pull request #3181 from AIFlowML/fix-multi-biome-02

fix: multi-biome-02

shakkernerd committed:
chore: remove multiple tsup package

shakkernerd committed:
Merge branch 'develop' into fix-plugin-0x

shakkernerd committed:
Merge pull request #3178 from AIFlowML/fix-plugin-0x

fix: plugin-0x

shakkernerd committed:
Merge branch 'develop' into fix-plugin-3g-gen

shakkernerd committed:
Merge pull request #3175 from AIFlowML/fix-plugin-3g-gen

fix: plugin-3g-generation

shakkernerd committed:
Merge branch 'develop' into fix-plugin-abstract

shakkernerd committed:
Merge pull request #3174 from AIFlowML/fix-plugin-abstract

fix: plugin-abstract

shakkernerd committed:
Merge branch 'develop' into fix-plugin-agentkit

shakkernerd committed:
Merge pull request #3172 from AIFlowML/fix-plugin-agentkit

fix: plugin-agentkit

shakkernerd committed:
Merge branch 'develop' into fix-plugin-akash-v1

shakkernerd committed:
Merge pull request #3171 from AIFlowML/fix-plugin-akash-v1

fix: plugin-akash

shakkernerd committed:
Merge branch 'develop' into fix-plugin-allors

shakkernerd committed:
Merge pull request #3169 from AIFlowML/fix-plugin-allors

fix: plugin-allora

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3165 from comfsrt/main

docs: fix typos in .md

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-ton
- plugin-trikon
- plugin-trustdb
- plugin-tts

Changes include:
- Add @biomejs/biome as devDependency
- Add lint and format scripts
- Replace existing ESLint configurations with Biome
- Maintain existing build and test scripts

ai16z-demirix committed:
feat (chore): plugin-coinmarketcap (#3134)

* plugin-coinmarketcap: vitest config

* plugin-coinmarketcap: package json

* plugin-coinmarketcap: test setup

* plugin-coinmarketcap: get price tests

* plugin-coinmarketcap: get price service tests

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

ai16z-demirix committed:
feat (chore): plugin-coingecko test config and coverage (#3124)

* plugin-coingecko: package json

* plugin-coingecko: vitest config

* plugin-coingecko: get price tests

* plugin-coingecko: test setup

* plugin-coingecko: get trending tests

* plugin-coingecko: top gainers losers tests

* plugin-coingecko: get market tests

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-thirdweb
- plugin-tee-verifiable-log
- plugin-tee-marlin
- plugin-tee-log

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-tee
- plugin-suno
- plugin-sui
- plugin-story

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-starknet
- plugin-stargaze
- plugin-squid-router
- plugin-spheron

AIFlowML committed:
Add Biome linting and formatting configuration to the following plugins:
- plugin-solana-v2
- plugin-solana-agent-kit
- plugin-solana
- plugin-sgx

AIFlowML committed:
Merge branch 'develop' into fix-multi-biome-07

tcm390 committed:
Merge branch 'main' into tcm/remove-duplicated-plugins

tcm390 committed:
Merge branch 'develop' into tcm/remove-duplicated-plugins

tcm390 committed:
Merge pull request #3126 from elizaOS/tcm/remove-duplicated-plugins

fix: remove duplicated plugins

tcm390 committed:
update provider-utils

tcm390 committed:
Merge branch 'develop' into tcm-update-provider-utils

tcm390 committed:
Merge pull request #3189 from elizaOS/tcm-update-provider-utils

fix: update provider-utils

tcm390 committed:
extract attribute from raw text instead of normalized json

tcm390 committed:
Merge pull request #3190 from elizaOS/tcm-correct-extract

fix: extract attribute from raw text instead of normalized json

samarth30 committed:
Merge branch 'develop' into fix-multi-biome-07

samarth30 committed:
Merge branch 'develop' into feat/plugin-ton-staking

samarth30 committed:
Merge branch 'develop' into feat/coingecko_advanced_more_endpoints

john-xina-p88 committed:
Merge branch 'develop' into develop

samarth30 committed:
Merge pull request #3170 from 0xCardinalError/feat/coingecko_advanced_more_endpoints

feat: coingecko advanced -  various pools by network

samarth30 committed:
Merge branch 'develop' into feat/plugin-ton-staking

samarth30 committed:
Merge pull request #3045 from edwin-finance/plugin-edwin

feat: Add edwin plugin to eliza

samarth30 committed:
Merge pull request #3096 from HMXOrg/develop

feat: plugin desk exchange

samarth30 committed:
Merge branch 'develop' into feat/plugin-ton-staking

maxime committed:
fix: make slack client able to download and upload attachments

tylermcwilliams commented:
Should swap the testnet staking contract to ton whales nominator #1 since their contract is verified
https://github.com/tylermcwilliams/eliza/commit/965e3d4f45d4faba920b28907fbc4d3e92a56e35

nilaysarma committed:
Update Twitter to X (Twitter)

Since Twitter is now rebranded to X, so I have changes made changes in the README file. It is now "X (Twitter)" and I kept the Twitter keyword since people still use the word Twitter.

tylermcwilliams committed:
chore/test: swapped testnet staking contract address with TON Whales nominator, changed getter in getPoolInfo

tylermcwilliams committed:
feat: add staking platform interface

tcm390 committed:
Merge branch 'develop' into 3196--Update-Twitter-to-X

tcm390 committed:
Merge pull request #3198 from nilaysarma/3196--Update-Twitter-to-X

docs: Update Twitter to X (Twitter)

tcm390 committed:
Merge branch 'develop' into fix/slack_download_upload_attachments

Marcofann committed:
Update index.ts

tylermcwilliams committed:
feat: added PlatformFactory and stakingPoolAddresses.json

tylermcwilliams committed:
feat: added tonWhales strategy

tylermcwilliams committed:
feat/test: StakingProvider getPoolInfo implements strategy pattern, changed pool info test description

shakkernerd committed:
Merge branch 'develop' into fix-multi-biome-07

shakkernerd committed:
Merge pull request #3186 from AIFlowML/fix-multi-biome-07

chore: add Biome configuration to Solana ecosystem plugins 07

gap-editor committed:
typo chat_2024-12-01.md

solono - solana

gap-editor committed:
typo chat_2024-12-03.md

vides - videos

gap-editor committed:
Update chat_2024-10-30.md

Docker composes - Docker Compose

shakkernerd committed:
Merge branch 'develop' into fix-plugin-anyone

shakkernerd committed:
Merge pull request #3107 from AIFlowML/fix-plugin-anyone

fix: plugin-anyone

YoungPhlo committed:
Merge branch 'develop' into docs/contributor-meetings

rferrari committed:
fix: think tag from venice

YoungPhlo committed:
docs: Add weekly contributor meeting notes (01-21 and 01-28)

gap-editor committed:
Update chat_2024-11-23.md

gap-editor committed:
typo chat_2024-12-02.md

YoungPhlo committed:
capitalized ElizaOS (for graphite)

Danyylka committed:
Update faq.md

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3207 from Danyylka/main

docs: Update faq.md

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3206 from gap-editor/main

chore: Fix Typos and Improve Consistency in Community Chat Logs

shakkernerd committed:
Merge branch 'develop' into docs/contributor-meetings

shakkernerd committed:
Merge pull request #3204 from YoungPhlo/docs/contributor-meetings

docs: Add weekly contributor meeting notes

shakkernerd committed:
Merge branch 'develop' into develop

shakkernerd committed:
Merge pull request #3203 from rferrari/develop

fix: think tag from venice

shakkernerd committed:
Merge branch 'develop' into fix/slack_download_upload_attachments

shakkernerd committed:
Merge pull request #3194 from maxime/fix/slack_download_upload_attachments

fix: Slack download upload attachments

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3166 from PixelPil0t1/main

chore: Update GitHub Actions workflows and documentation

shakkernerd committed:
Merge branch 'develop' into fix/fix

shakkernerd committed:
Merge pull request #3164 from Marcofann/fix/fix

chore: Standardization of Security Check Identifiers across GoPlus plugin

shakkernerd committed:
Merge branch 'develop' into docker-pnpm-version

shakkernerd committed:
Merge pull request #3158 from v1xingyue/docker-pnpm-version

fix: Update pnpm version during Docker build

shakkernerd committed:
Merge branch 'develop' into fix-readme-ja

shakkernerd committed:
Merge pull request #3153 from You-saku/fix-readme-ja

Fix README_JA.md (add unwritten text and fix typo)

shakkernerd committed:
Merge branch 'develop' into some-fix

shakkernerd committed:
chore: conditional logging for server starting on a different port

shakkernerd committed:
Merge pull request #3141 from v1xingyue/some-fix

chore: some fix after v0.1.9

silasneo committed:
feat: Trump character but tweets in Spanish by default (#3119)

* Update README.md

small note at the top of readme to highlight contribution via this fork

* custom character file

* Trump in Latin American Spanish

Modified Trump character file to have the agent speak in Latin Spanish by default.
The agent will respond in any language if specifically asked to respond in that language in the user input.
This file can serve as a template for creating agents who speak a specific language by default for a given language audience.

* Update README.md

* Delete characters/uncleugo4.character.json

* Adding spanish_trump.character.json

Many people are not native English speakers & a template was needed to make it easier for anyone to create an agent that speaks other languages by default. 
Modified Trump character file to have the agent speak in Latin Spanish by default. 
This file can serve as a template for creating agents who speak a specific language by default for a given language audience.
The agent will respond in any language if specifically asked to respond in that language in the user input.

* Update spanish_trump.character.json

minor correction for duplicated adjectives

* Update spanish_trump.character.json

* removed duplicate adjectives

* Update package.json

* Update spanish_trump.character.json

Removed some English in bio for consistency

---------

Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

azep-ninja committed:
optimizations & fixes.

elpulpo0 committed:
feat: Add CREATE_POOL action in MultiversX plugin

Fallengirl committed:
Update chat_2024-11-17.md

elpulpo0 committed:
feat: update constants in MultiversX plugin

mikirov committed:
feat: nft actions

odilitime committed:
Merge pull request #3211 from mikirov/feat/ton-plugin-nft-management

feat: TON Plugin: NFT collection, item creation, metadata change and transfer actions

odilitime committed:
Merge pull request #3210 from Fallengirl/patch-1

docs: rename chat_2024-11-17.md

odilitime committed:
Merge pull request #3209 from elpulpo0/develop

feat: Add CREATE_POOL action in MultiversX plugin

odilitime committed:
Merge branch 'develop' into fix-quickintel-plugin

tcm390 committed:
remove duplicated dependencies

tcm390 committed:
Merge pull request #3215 from elizaOS/tcm-remove-duplicated-dependencies

fix: remove duplicated dependencies

shakkernerd committed:
Merge branch 'develop' into fix-quickintel-plugin

shakkernerd committed:
Merge pull request #3208 from azep-ninja/fix-quickintel-plugin

fix: quick-intel plugin optimizations & fixes.

tcm390 committed:
add ENABLE_TWITTER_POST_GENERATION config

tcm390 committed:
add ENABLE_TWITTER_POST_GENERATION config

tcm390 committed:
add ENABLE_TWITTER_POST_GENERATION

shakkernerd committed:
Merge pull request #3219 from elizaOS/tcm-twitter-disable-post

feat: Add configuration for enabling/disabling Twitter post generation

cypherpepe committed:
Update broken link awesome-eliza.md

cypherpepe committed:
Update broken link README.md

cypherpepe committed:
Update broken link README.md

cypherpepe committed:
Update broken link profiles.mdx

cypherpepe committed:
Update broken link chat_2024-11-16.md

tylermcwilliams committed:
chore: changed platform strategy interface to return ton MessageRelaxed

tylermcwilliams committed:
feat: implemented staking and unstaking message creation in tonWhales strategy

tylermcwilliams committed:
feat: staking provider now uses strategy pattern for stake and unstake

tylermcwilliams committed:
feat: added hipo strategy for Hipo liquid staking

ai16z-demirix committed:
plugin-coflux: package json

ai16z-demirix committed:
plugin-coflux: vitest config

ai16z-demirix committed:
plugin-coflux: setup

ai16z-demirix committed:
plugin-coflux: bridge transfer tests

ai16z-demirix committed:
plugin-coflux: transfer tests

ai16z-demirix committed:
plugin-cronos: vitest config

ai16z-demirix committed:
plugin-cronos: package json

ai16z-demirix committed:
plugin-cronos: setup

ai16z-demirix committed:
plugin-cronos: balance tests

ai16z-demirix committed:
plugin-cronos: transfer tests

madschristensen99 committed:
Delete packages/plugin-lit/src/providers directory

madschristensen99 committed:
Delete packages/plugin-lit/src/config directory

madschristensen99 committed:
Delete packages/plugin-lit/src/actions/sendSol.ts

madschristensen99 committed:
Delete packages/plugin-lit/src/actions/sendEth.ts

madschristensen99 committed:
Delete packages/plugin-lit/src/actions/sendUSDC.ts

madschristensen99 committed:
Delete packages/plugin-lit/src/actions/index.ts

madschristensen99 committed:
Delete packages/plugin-lit/src/types directory

madschristensen99 committed:
Delete packages/plugin-lit/pkp-setup.md

madschristensen99 committed:
Delete packages/plugin-lit/.npmignore

madschristensen99 committed:
Update README.md

shakkernerd committed:
Merge pull request #3250 from ai16z-demirix/tests/plugin-cronos

feat (chore): plugin-cronos test setup and coverage

shakkernerd committed:
Merge branch 'develop' into tests/plugin-conflux

shakkernerd committed:
Merge pull request #3247 from ai16z-demirix/tests/plugin-conflux

feat (chore) plugin conflux: test config and coverage

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3240 from cypherpepe/main

fix: broken links in documentation

AIFlowML committed:
Add Biome linting and formatting configuration to client-alexa:
- Add @biomejs/biome as devDependency
- Add lint and format scripts
- Replace existing ESLint configuration
- Standardize tsup dependency version

shakkernerd committed:
Merge branch 'develop' into main

shakkernerd committed:
Merge pull request #3251 from madschristensen99/main

chore: remove unnecessary provider and transfer code

shakkernerd committed:
Merge branch 'develop' into fix-client-alexa

shakkernerd committed:
Merge pull request #3255 from AIFlowML/fix-client-alexa

fix: client-alexa

renovate[bot] committed:
fix(deps): update dependency vitest [security] (#3254)

Co-authored-by: renovate[bot] <29139614+renovate[bot]@users.noreply.github.com>
Co-authored-by: Shakker Nerd <165377636+shakkernerd@users.noreply.github.com>

tcm390 committed:
handle invalid json format

tcm390 committed:
trim input

tcm390 committed:
use raw text to normalized

tcm390 committed:
Merge branch 'develop' into tcm-handle-invalid-json

tcm390 committed:
fix install

tcm390 committed:
Merge pull request #3261 from elizaOS/tcm-fix-install

fix: pnpm install

tcm390 committed:
fix http dynamic require issue

tcm390 committed:
Merge branch 'develop' into tcm/fix-plugin-ton

tcm390 committed:
Merge pull request #3262 from elizaOS/tcm/fix-plugin-ton

fix: plugin-ton Dynamic require of "http" is not supported

esen committed:
Fix PG query: Use only 'text' subfield name, field name is not needed here

nilaysarma committed:
add github issues link

tylermcwilliams committed:
chore: fixed nativeStakingProvider, stake action now instantiates StakingProvider thorugh initStakingProvider

johntad110 committed:
Fix broken links on contributing.md

lalalune committed:
fix develop build issue

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

lalalune committed:
Add chracters

lalalune committed:
remove minified code block

lalalune committed:
Merge pull request #3269 from johntad110/fix-broken-links

Fix broken links on contributing.md

lalalune committed:
Merge pull request #3268 from nilaysarma/add-link-to-issues

docs: Add GitHub issues link to CONTRIBUTING.md file

lalalune committed:
Merge pull request #3264 from esen/fix_pg_memory_query

Fix PG query: Use only 'text' subfield name, field name is not needed here

lalalune committed:
Merge branch 'develop' into tcm-handle-invalid-json

lalalune committed:
Merge pull request #3258 from elizaOS/tcm-handle-invalid-json

fix: handle invalid json

lalalune committed:
Merge branch 'develop' of http://github.com/ai16z/eliza into HEAD

mikirov committed:
Merge branch 'develop' into feat/plugin-ton-staking

tylermcwilliams committed:
chore: modified action descriptors

tylermcwilliams committed:
fix: getStrategy can get HIPO strategy type

tylermcwilliams committed:
fix: nested hiposdk files in workspace as workaround for module cycling problems

tylermcwilliams committed:
feat: modified stakingPlatform interface, we now construct strategies with TonClient and WalletProvider instance

tylermcwilliams committed:
Update dependencies

tylermcwilliams committed:
Merge remote-tracking branch 'refs/remotes/mikirov/feat/plugin-ton-staking' into feat/plugin-ton-staking

tylermcwilliams committed:
feat: added getPendingWithdraw to Startegy interface, updated strategies implementations

odilitime commented:
Hello,

We are changing our plugin development strategy to be more scalable. We have moved the plugins out into their own repos and we're looking for people to either maintain those or own them on their own Github.

Please make your code changes to this repo: https://github.com/elizaos-plugins/plugin-ton

If you'd like to be a maintainer,  file an issue in the plugin repo and join our Discord https://discord.gg/elizaos to coordinate.

If you'd like to control the plugin on your own Github, please add an issue to the plugin repo pointing to your repo, and add a modification to the registry. Submit a PR to edit the registry here: https://github.com/elizaos-plugins/registry

Closing this PR for now. Let us know if you have any questions.
